calculator.js

```diff
@@ -1,126 +1,134 @@
-function Calculator() {
-    'use strict';
-
-    let newInput = "0";
-    let oldInput = "";
-    let result = "";
-    this.input = ko.observable(newInput);
-    let operation = "";
-    let history = [];
-
-    //--------ADD NUMBER---------
-    this.addNumber = (number) => () => {
-        if (newInput === "0") {
-            newInput = number.toString();
+(function (factory) {
+    if (typeof require === 'function' && typeof exports === 'object' && typeof module === 'object') {
+        module.exports = factory(require('knockout'));

```

Comment:
This line of code will run if the code is running inside Node JS. If that is the case, we pass knockout to our factory by doing `factory(require('knockout'))`. We also export the result with `module.exports = ...`.